docs/src/testing/using-cheatcodes.md

```diff
@@ -19,101 +19,26 @@ using [cheatcodes](../appendix/cheatcodes.md).
 In this tutorial, we will be using the following Starknet contract:
 
 ```rust
-#[starknet::interface]
-trait IHelloStarknet<TContractState> {
-    fn increase_balance(ref self: TContractState, amount: felt252);
-    fn get_balance(self: @TContractState) -> felt252;
-    fn get_block_number_at_construction(self: @TContractState) -> u64;
-    fn get_block_timestamp_at_construction(self: @TContractState) -> u64;
-}
-
-#[starknet::contract]
-mod HelloStarknet {
-
-    use box::BoxTrait;
-    use starknet::{Into, get_caller_address};
-
-    #[storage]
-    struct Storage {
-        balance: felt252,
-        blk_nb: u64,
-        blk_timestamp: u64,
-    }
-
-    #[constructor]
-    fn constructor(ref self: ContractState) {
-        // store the current block number
-        self.blk_nb.write(starknet::get_block_info().unbox().block_number);
-        // store the current block timestamp
-        self.blk_timestamp.write(starknet::get_block_info().unbox().block_timestamp);
-    }
-
-    #[abi(embed_v0)]
-    impl IHelloStarknetImpl of super::IHelloStarknet<ContractState> {
-        // Increases the balance by the given amount.
-        fn increase_balance(ref self: ContractState, amount: felt252) {
-            assert_is_allowed_user();
-            self.balance.write(self.balance.read() + amount);
-        }
-        // Gets the balance.
-        fn get_balance(self: @ContractState) -> felt252 {
-            self.balance.read()
-        }
-        // Gets the block number
-        fn get_block_number_at_construction(self: @ContractState) -> u64 {
-            self.blk_nb.read()
-        }
-        // Gets the block timestamp
-        fn get_block_timestamp_at_construction(self: @ContractState) -> u64 {
-            self.blk_timestamp.read()
-        }
-    }
-
-    fn assert_is_allowed_user() {
-        // checks if caller is '123'
-        let address = get_caller_address();
-        assert(address.into() == 123, 'user is not allowed');
-    }
-}
+{{#include ../../listings/snforge_overview/crates/using_cheatcodes/src/lib.cairo}}
 ```
 
-Please note that this contract example is a continuation of the same contract as in the [Testing Smart Contracts](./testing.md) page.
-
 ## Writing Tests
 
 We can try to create a test that will increase and verify the balance.
 
 ```rust
-#[test]
-fn call_and_invoke() {
-    // ...
-
-    let balance = dispatcher.get_balance();
-    assert(balance == 0, 'balance == 0');
-
-    dispatcher.increase_balance(100);
-
-    let balance = dispatcher.get_balance();
-    assert(balance == 100, 'balance == 100');
-}
+{{#include ../../listings/snforge_overview/crates/using_cheatcodes/tests/caller_address/failing.cairo}}
 ```
 
-However, when running this test, we will get a failure with a message
+This test passes, which means that `increase_balance` method panics with expected message.
```

Comment:
Why was this changed to should panic?

Reply:
We cannot have panicking tests snce we want to verify our listings using `scarb test`.
Here I made a mistake - `#[should_panic]` attribute is not shown and the test is described as panicking, so I should restore the old text.

Reply:
We cannot have panicking tests since we want to verify Cairo listings using `scarb test`. That's why some tests are marked `#[should_panic]` and some use safe dispatchers.
Also a few tests in the first chapter introducing to testing are presented as failing but in fact they have a hidden `#[should_panic]` not appearing in the text.

Reply:
I've looked at the whole document again and I think this is confusing to have it pass. If that's necessary you can add hidden `should_panic` but we should be showing to the users that this fails so it's obvious what's the point of using a cheat there.